core/src/cell.rs

```diff
@@ -341,11 +342,58 @@ impl HeaderChecker for BlockHeadersChecker {
 pub enum UnresolvableError {
     /// The header is not in main chain
     InvalidHeader(H256),
+    /// Invalid dep group cell data length
+    InvalidDepGroup(OutPoint),
     Dead(OutPoint),
     Unknown(Vec<OutPoint>),
     OutOfOrder(OutPoint),
 }
 
+pub fn parse_dep_group_data(data: Bytes) -> Result<Vec<OutPoint>, usize> {
+    // tx hash (32 bytes) + output index (4 bytes)
+    const OUT_POINT_LEN: usize = mem::size_of::<H256>() + mem::size_of::<u32>();
+
+    if data.is_empty() || data.len() % OUT_POINT_LEN != 0 {
+        return Err(data.len());
+    }
+
+    let dep_len = data.len() / OUT_POINT_LEN;
+    let mut out_points = Vec::with_capacity(dep_len);
+    for dep_idx in 0..dep_len {
+        let item_start = dep_idx * OUT_POINT_LEN;
+        let item_end = (dep_idx + 1) * OUT_POINT_LEN;
+        out_points.push(
+            OutPoint::from_group_data(&data[item_start..item_end])
+                .expect("parse group item data failed"),
```

Comment:
should return Err even if we know the length is correct.

Reply:
We already checked the length above, so it can not be error.

Reply:
They have only a fine distinction between them. If we refactor the from_group_data, it's very easily overlooked here, this `expect` depends on the implementation of the `from_group_data`.It's not serious.